src/Chapter3.hs

```diff
@@ -560,22 +701,50 @@ introducing extra newtypes.
 🕯 HINT: if you complete this task properly, you don't need to change the
     implementation of the "hitPlayer" function at all!
 -}
+newtype Attack = Attack
+    { attackPts :: Int
+    } deriving (Show)
+
+newtype Health = Health
+    { healthPts :: Int
+    } deriving (Show)
+
+newtype Armor = Armor
+    { armorPts :: Int
+    } deriving (Show)
+
+newtype Dexterity = Dexterity
+    { dexterityPts :: Int
+    } deriving (Show)
+
+newtype Strength = Strength
+    { strengthPts :: Int
+    } deriving (Show)
+
+newtype Damage = Damage
+    { damagePts :: Int
+    } deriving (Show)
+
+newtype Defense = Defense
+  { defensePts :: Int
+  } deriving (Show)   
+
 data Player = Player
-    { playerHealth    :: Int
-    , playerArmor     :: Int
-    , playerAttack    :: Int
-    , playerDexterity :: Int
-    , playerStrength  :: Int
-    }
+    { playerHealth    :: Health
+    , playerArmor     :: Armor
+    , playerAttack    :: Attack
+    , playerDexterity :: Dexterity
+    , playerStrength  :: Strength
+    } deriving (Show)
 
-calculatePlayerDamage :: Int -> Int -> Int
-calculatePlayerDamage attack strength = attack + strength
+calculatePlayerDamage :: Attack -> Strength -> Damage
+calculatePlayerDamage attack strength = Damage ((attackPts attack) + (strengthPts strength))
 
-calculatePlayerDefense :: Int -> Int -> Int
-calculatePlayerDefense armor dexterity = armor * dexterity
+calculatePlayerDefense :: Armor -> Dexterity -> Defense
+calculatePlayerDefense armor dexterity = Defense ((armorPts armor) * (dexterityPts dexterity))
 
-calculatePlayerHit :: Int -> Int -> Int -> Int
-calculatePlayerHit damage defense health = health + defense - damage
+calculatePlayerHit :: Damage -> Defense -> Health -> Health
+calculatePlayerHit damage defense health = Health ((healthPts health) + (defensePts defense) - (damagePts damage))
 
 -- The second player hits first player and the new first player is returned
 hitPlayer :: Player -> Player -> Player
```

Comment:
Nice work on newtypes! :medal_sports: Newtypes are one very important feature in Haskell :+1:
You can read more about the newtype pattern in our blog post: https://kowainik.github.io/posts/haskell-mini-patterns#newtype